safe-global/web-core
# feat: Add a settings page to change env vars in runtime

## What it solves

Resolves #1505 

## ToDo

- [x] Add page outside of safe to edit form `/environment-variables`
- [x] Add field to change RPC URI
- [x] Add field to change Tenderly URL
- [x] Add Hint/Warning in the sidebar if custom values are set – need design @liliiaorlenko 
- [x] Show links in the form explaining what those vars are
- [x] Track config changes
- [ ] Potentially refactor how Tenderly/RPC URLs are formed from the config


## How this PR fixes it

- Adds a new form to change the RPC URL, Tenderly URL and Client gateway URL in runtime
- Shows a hint in the sidebar if the default values were changed
- Displays the form on a stand-alone page

## How to test it

1. Open a safe
2. Navigate to Settings
3. Observe a new page called Environment variables
4. Navigate to that page
5. Observe input fields for RPC provider and Tenderly
6. Add a different RPC provider URL and save the form
7. Observe the page reloading
8. Observe a visible hint in the sidebar that links to the form page
9. Observe that all RPC requests go through the new URL

## Analytics changes

- Tracks the number of times the form has been saved

## Screenshots
<img width="1268" alt="Screenshot 2023-01-18 at 10 47 31" src="https://user-images.githubusercontent.com/5880855/213138497-0d747ee4-2840-40f5-b3e6-198e28692c86.png">
<img width="402" alt="Screenshot 2023-01-13 at 12 24 25" src="https://user-images.githubusercontent.com/5880855/212309626-541e24f2-1e84-4de2-ad70-00fc4db5208a.png">
